ompi/mca/pml/ob1/pml_ob1_accelerator.c

```diff
@@ -178,6 +179,101 @@ int mca_pml_ob1_progress_one_htod_event(struct mca_btl_base_descriptor_t **frag)
     return 0;
 }
 
+int mca_pml_ob1_record_dtoh_event(char *msg, struct mca_btl_base_descriptor_t *frag)
+{
+    int result;
+
+    if (0 == strcmp(accelerator_base_selected_component.base_version.mca_component_name, "null")) {
+        return 0;
+    }
+
+    /* First make sure there is room to store the event.  If not, then
+     * return an error.  The error message will tell the user to try and
+     * run again, but with a larger array for storing events. */
+    OPAL_THREAD_LOCK(&pml_ob1_accelerator_dtoh_lock);
+    if (accelerator_event_dtoh_num_used == accelerator_event_max) {
+        opal_output_verbose(1, mca_pml_ob1_output, "Out of event handles. Max: %d. Suggested to rerun with new max with mca mpi_common_accelerator_event_max %d.",
+                            accelerator_event_max, accelerator_event_max + 100);
+        return OPAL_ERR_OUT_OF_RESOURCE;
+    }
+
+    if (accelerator_event_dtoh_num_used > accelerator_event_dtoh_most) {
+        accelerator_event_dtoh_most = accelerator_event_dtoh_num_used;
+        /* Just print multiples of 10 */
+        if (0 == (accelerator_event_dtoh_most % 10)) {
+            opal_output_verbose(20, mca_pml_ob1_output, "Maximum DtoH events used is now %d",
+                                accelerator_event_dtoh_most);
+        }
+    }
+
+    result = opal_accelerator.record_event(&accelerator_event_dtoh_array[accelerator_event_dtoh_first_avail], &dtoh_stream);
+    if (OPAL_UNLIKELY(OPAL_SUCCESS != result)) {
+        opal_output_verbose(1, mca_pml_ob1_output, "Event Record failed.");
+        OPAL_THREAD_UNLOCK(&pml_ob1_accelerator_dtoh_lock);
+        return OPAL_ERROR;
+    }
+    accelerator_event_dtoh_frag_array[accelerator_event_dtoh_first_avail] = frag;
+
+    /* Bump up the first available slot and number used by 1 */
+    accelerator_event_dtoh_first_avail++;
+    if (accelerator_event_dtoh_first_avail >= accelerator_event_max) {
+        accelerator_event_dtoh_first_avail = 0;
+    }
+    accelerator_event_dtoh_num_used++;
+
+    OPAL_THREAD_UNLOCK(&pml_ob1_accelerator_dtoh_lock);
+    return OPAL_SUCCESS;
+}
+
+/**
+ * Progress any dtoh event completions.
+ */
+int mca_pml_ob1_progress_one_dtoh_event(struct mca_btl_base_descriptor_t **frag)
+{
+    int result;
+
+    if (0 == strcmp(accelerator_base_selected_component.base_version.mca_component_name, "null")) {
+        return 0;
```

Comment:
Shoulld that be `OPAL_SUCCESS`? I wonder if the return values should be `OMPI_*` variants in this component...